test/utils/test_zenith_angle.py

```diff
@@ -61,3 +66,122 @@ def test_zenith_angle_array():
     lon = np.array([0.0])[None, None, :]
     out = cos_zenith_angle_from_timestamp(timestamp, lon, lat)
     assert out.shape == (3, 2, 1)
+
+
+@pytest.mark.parametrize(
+    "t",
+    [
+        datetime(2020, 7, 6, 9, 0, 0),
+        datetime(2000, 1, 1, 12, 0, 0),
+        datetime(2000, 7, 1, 12, 0, 0),
+        datetime(2000, 7, 1, 12, 0, 0, tzinfo=utc),
+    ],
+)
+def test_timestamp_to_julian_centuries(t):
+    a = _datetime_to_julian_century(t)
+    b = _timestamp_to_julian_century(t.replace(tzinfo=utc).timestamp())
+    assert a == b
+
+
+def test_toa():
+    t = datetime(2000, 7, 1, 12, 0, 0, tzinfo=utc).timestamp()
+    lat, lon = 0.0, 0.0
+    ans = toa_incident_solar_radiation_accumulated(t, lat, lon)
+    assert ans >= 0
+
+
+@pytest.skip(
+    "This test requires CDS API and manual review of the tisr.png "
+    "and line .png. Consider running if modifying the tisr code."
+)
+def test_tisr_matches_cds():
+    import os
+
+    import cdsapi
+    import matplotlib.pyplot as plt
+
+    grid = (1, 1)
+    area = (90, -180, -90, 180)
+    format = "netcdf"
+
+    client = cdsapi.Client()
+
+    if not os.path.isfile("out.nc"):
+        client.retrieve(
+            "reanalysis-era5-single-levels",
+            {
+                "product_type": "reanalysis",
+                "variable": [212],
+                "year": [2018, 2019, 2020],
+                "month": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
+                "day": 1,
+                "time": "00:00",
+                "area": area,
+                "grid": grid,
+                "format": format,
+            },
+            "out.nc",
+        )
+
+    import xarray
+
+    ds = xarray.open_dataset("out.nc")
+
+    def plot_comparison(ds):
+
+        t = ds.time.values.astype("datetime64[s]").astype("int64")
+
+        # lat = np.linspace(-90, 90, 721)[:, None]
+        # lon = np.linspace(0, 360, 1440, endpoint=False)[None, :]
+        t, lat, lon = np.meshgrid(t, ds.latitude, ds.longitude, indexing="ij")
+
+        tisr = toa_incident_solar_radiation_accumulated(t, lat, lon)
+
+        fig, axs = plt.subplots(
+            t.shape[0], 3, figsize=(15, 30), sharex=True, sharey=True
+        )
+        for i in range(t.shape[0]):
+            a, b, c = axs[i].flat
+            im = a.pcolormesh(ds.longitude, ds.latitude, ds.tisr[i] / 3600)
+            plt.colorbar(im)
+            im = b.pcolormesh(ds.longitude, ds.latitude, tisr[i] / 3600)
+            plt.colorbar(im)
+            im = c.pcolormesh(ds.longitude, ds.latitude, (tisr[i] - ds.tisr[i]) / 3600)
+            plt.colorbar(im)
+
+            time_str = ds.time[i].dt.strftime("%Y-%m-%d %H:%M:%S").item()
+            a.set_ylabel(time_str)
+
+            if i == 0:
+
+                a.set_title("ERA5")
+                b.set_title("Computed")
+                c.set_title("Difference")
+                fig.suptitle("Solar irradiance (W/m^2)")
+
+    plot_comparison(ds.sel(time="2018"))
+    plt.savefig("tisr.png")
```

Comment:
Would be good to convert this into a test that checks for accuracy up to a certain threshold. Can keep the plotting code here for future use, but branch it off using a verbosity flag. Something like https://github.com/NVIDIA/modulus/blob/main/test/distributed/test_distributed_fft.py#L79